pkg/platform/runtime/buildexpression/merge/merge.go

```diff
@@ -56,22 +58,26 @@ func Merge(exprA *buildexpression.BuildExpression, exprB *buildexpression.BuildE
 // expressions.
 // This is only possible if the two build expressions differ ONLY in requirements.
 func isAutoMergePossible(exprA *buildexpression.BuildExpression, exprB *buildexpression.BuildExpression) bool {
-	jsonA, err := getJsonMinusRequirements(exprA)
+	jsonA, err := getComparableJson(exprA)
 	if err != nil {
 		multilog.Error("Unable to get buildexpression minus requirements: %v", errs.JoinMessage(err))
 		return false
 	}
-	jsonB, err := getJsonMinusRequirements(exprB)
+	jsonB, err := getComparableJson(exprB)
 	if err != nil {
 		multilog.Error("Unable to get buildxpression minus requirements: %v", errs.JoinMessage(err))
 		return false
 	}
-	return reflect.DeepEqual(jsonA, jsonB)
+	logging.Debug("Checking for possibility of auto-merging build expressions")
+	logging.Debug("JsonA: %v", jsonA)
+	logging.Debug("JsonB: %v", jsonB)
```

Comment:
Guessing this was for local debugging? Or you want to keep this verbose for a while since this'll likely need some further debugging?

Reply:
The latter. I've only ID'ed 3 cases in which build expressions may be different, but equal (as pointed out in DX-1939): different key-value order in objects, different list order, and different "at_time" values. There could be more, and I'd like to have this easily accessible for now. All the JSON is on one line and it's not too bad in terms of size, so it won't bloat up the logs like graphql queries and responses do.